### PR TITLE
Restructure top level folders

### PR DESCRIPTION
**Pull request by abhi-mosaic**
TODO:

- [x] READMEs for every top level folder
- [x] fix most tests
- future PR: fix tests that test tscripts (@dakinggg I could use your help)
- [x] fix any documentation and MCLI yamls whose paths have changed
- [x] look into if there is a cleaner presentation than having `assets`, `tests/`, bunch of other files at toplevel (@alextrott16)
- [x] ~figure out what to do about `data/finetuning/tasks.py`~

For best viewing experience, I recommend looking at the branch in Github UI: https://github.com/mosaicml/llm-foundry/tree/abhi/restruct

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,186 +46,48 @@ To get started, clone this repo and install the requirements:
 
 ```bash
 git clone https://github.com/mosaicml/llmfoundry.git
```
Comment (alextrott16) [inline README.md:48]:
```suggestion
git clone https://github.com/mosaicml/llm-foundry.git
```

File: README.md [hunk 2]
```diff
@@ -56,186 +46,48 @@ To get started, clone this repo and install the requirements:
...
-pip install -e ".[gpu]"  # or pip install -e . if no NVIDIA GPU
 cd llmfoundry
```
Comment (alextrott16) [inline README.md:49]:
```suggestion
cd llm-foundry
```

File: train/README.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+As a quickstart, here is how to prepare the [C4: Colossal, Cleaned, Common Crawl dataset](https://huggingface.co/datasets/c4).
+We first convert the dataset from its native format (a collection of zipped JSONs)
+to MosaicML's streaming dataset format (a collection of binary `.mds` files).
+Once in `.mds` format, we can store the dataset in a central location (filesystem, S3, GCS, etc.)
+and stream the data to any compute cluster, with any number of devices, and any number of CPU workers, and it all ~ just works ~ .
+You can read more about the benefits of using mosaicml-streaming [here](https://streaming.docs.mosaicml.com/en/stable/).
+
+NOTE: If you only want to profile these LLMs, we recommend that you **download and prepare the `train_small` and `val` splits**,
+and skip the full `train` split. You'll just need to replace `split: train` with `split: train_small` in your run YAML's dataloader config, [e.g. here](./llmfoundry/yamls/mosaic_gpt/125m.yaml).
```
Comment (vchiley) [inline train/README.md:24]:
```suggestion
and skip the full `train` split. You'll just need to replace `split: train` with `split: train_small` in your run YAML's dataloader config, [e.g. here](./llmfoundry/train/yamls/mosaic_gpt/125m.yaml).
```

File: train/README.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+To run pretraining, you'll need to make yourself a copy of a pretraining dataset. Check out the `llm-foundry/data_prep` folder for detailed instructions.
+
+As a quickstart, here is how to prepare the [C4: Colossal, Cleaned, Common Crawl dataset](https://huggingface.co/datasets/c4).
+We first convert the dataset from its native format (a collection of zipped JSONs)
+to MosaicML's streaming dataset format (a collection of binary `.mds` files).
+Once in `.mds` format, we can store the dataset in a central location (filesystem, S3, GCS, etc.)
+and stream the data to any compute cluster, with any number of devices, and any number of CPU workers, and it all ~ just works ~ .
+You can read more about the benefits of using mosaicml-streaming [here](https://streaming.docs.mosaicml.com/en/stable/).
+
+NOTE: If you only want to profile these LLMs, we recommend that you **download and prepare the `train_small` and `val` splits**,
```
Comment (vchiley) [inline train/README.md:23]:
```suggestion
NOTE: If you only want to profile these LLMs, we recommend that you **download and prepare the `train_small` and `val_small` splits**,
```

Also requires an update [here](https://github.com/mosaicml/llm-foundry/blob/abhi/restruct/data_prep/convert_dataset.py#L39)
(we have val_small, but didn't add it as an option)

File: train/README.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+and skip the full `train` split. You'll just need to replace `split: train` with `split: train_small` in your run YAML's dataloader config, [e.g. here](./llmfoundry/yamls/mosaic_gpt/125m.yaml).
+You can also accomplish this in your CLI command like so: `composer train.py ... train_loader.dataset.split=train_small`
```
Comment (vchiley) [inline train/README.md:25]:
```suggestion
You can also accomplish this in your CLI command like so: `composer train.py ... train_loader.dataset.split=train_small eval_loader.dataset.split=val_small`
```

**Comment by abhi-mosaic** [comment]
Moved scripts to a `scripts/` folder. Fixed a lot of references but probably not all of them. Will leave to tomorrow's bug bash (and relevant owners) to fix READMEs for each subfolder

**Review by vchiley — APPROVED** [review]
❤️

**Comment by abhi-mosaic** [comment]
Merged as it's getting late, and the pytest-cpu tests are all passing, just something wrong with `tests/conftest.py`

**Pull request merged**